fanout/join_group_fanout.js

```diff
@@ -0,0 +1,202 @@
+const sumOf = arr => arr.reduce((sum, n) => sum + n, Number());
+
+/** Join a fanout group
+
+  {
+    ask: <Ask Function>
+    code: <Group Invite Code String>
+    [inputs]: [<Utxo Outpoint String>]
+    [is_selecting_utxos]: <Interactively Select UTXOs to Spend Bool>
+    lnd: <Authenticated LND API Object>
+    logger: <Winston Logger Object>
+    max_rate: <Max Opening Chain Fee Tokens Per VByte Fee Rate Number>
+    output_count: <Output Count Number>
+  }
+
+  @returns via cbk or Promise
+  {
+    transaction_id: <Fanout Funding Transaction Id Hex String>
+  }
+*/
+module.exports = (args, cbk) => {
+  return new Promise((resolve, reject) => {
+    return asyncAuto({
+      // Check arguments
+      validate: cbk => {
+        if (!args.ask) {
+          return cbk([400, 'ExpectedAskFunctionToJoinGroupFanout']);
+        }
+
+        if (!isArray(args.inputs)) {
+            return cbk([400, 'ExpectedArrayOfUtxosToJoinGroupFanout']);
+        }
+
+        if (!isCode(args.code)) {
+          return cbk([400, 'ExpectedValidJoinCodeToJoinGroupFanout']);
+        }
+
+        if (!args.inputs.length && !args.is_selecting_utxos) {
+          return cbk([400, 'ExpectedToSelectUtxosToJoinGroupFanout']);
+        }
+
+        if (!args.lnd) {
+          return cbk([400, 'ExpectedAuthenticatedLndToJoinGroupFanout']);
+        }
+
+        if (!args.logger) {
+          return cbk([400, 'ExpectedWinstonLoggerToJoinGroupFanout']);
+        }
+
+        if (!args.max_rate) {
+          return cbk([400, 'ExpectedMaxOpeningFeeRateToJoinGroupFanout']);
+        }
+
+        if (!args.output_count || !isNumber(args.output_count)) {
+          return cbk([400, 'ExpectedOutputCountToJoinGroupFanout']);
+        }
+
+        return cbk();
+      },
+
+      // Get methods to confim partial signing is supported
+      getMethods: ['validate', ({}, cbk) => getMethods({lnd: args.lnd}, cbk)],
+
+      // Get UTXOs to use for input selection and final fee rate calculation
+      getUtxos: ['validate', ({}, cbk) => getUtxos({lnd: args.lnd}, cbk)],
+
+      // Make sure that partially signing a PSBT is valid
+      confirmSigner: ['getMethods', ({getMethods}, cbk) => {
+        if (!getMethods.methods.find(n => n.endpoint === signPsbtEndpoint)) {
+          return cbk([400, 'ExpectedLndSupportingPartialPsbtSigningToJoin']);
+        }
+
+        return cbk();
+      }],
+
+      // Select inputs to spend
+      utxos: ['getUtxos', ({getUtxos}, cbk) => {
+        // Exit early when UTXOs are all specified already
+        if (!!args.inputs.length) {
+          return cbk(null, args.inputs);
+        }
+
+        // Exit early when not selecting UTXOs interactively
+        if (!args.is_selecting_utxos) {
+          return cbk(null, []);
+        }
+
+        // Only selecting confirmed utxos is supported
+        const utxos = getUtxos.utxos
+        .filter(n => !!n.confirmation_count)
+        .filter(n => n.address_format !== nestedSegWitAddressFormat);
```

Comment:
I think this should really check for whitelisted output types to be safe against future output types that are unknown and therefore won't be supported, so like p2tr or p2wpkh